relayer/chains/cosmos/tx.go

```diff
@@ -649,22 +669,260 @@ func (cc *CosmosProvider) buildMessages(
 	// Build the transaction builder
 	txb, err := txf.BuildUnsignedTx(cMsgs...)
 	if err != nil {
-		return nil, 0, sdk.Coins{}, err
+		return nil, nil, err
 	}
+	return &txf, txb, nil
+}
 
-	if err = tx.Sign(txf, txSignerKey, txb, false); err != nil {
-		return nil, 0, sdk.Coins{}, err
+func convertAddress(addrString string) (*common.Address, error) {
+	cfg := sdk.GetConfig()
+	var addr []byte
+	// try hex, then bech32
+	switch {
+	case common.IsHexAddress(addrString):
+		addr = common.HexToAddress(addrString).Bytes()
+	case strings.HasPrefix(addrString, cfg.GetBech32ValidatorAddrPrefix()):
+		addr, _ = sdk.ValAddressFromBech32(addrString)
+	case strings.HasPrefix(addrString, cfg.GetBech32AccountAddrPrefix()):
+		addr, _ = sdk.AccAddressFromBech32(addrString)
+	default:
+		return nil, fmt.Errorf("expected a valid hex or bech32 address (acc prefix %s), got '%s'", cfg.GetBech32AccountAddrPrefix(), addrString)
 	}
+	to := common.BytesToAddress(addr)
+	return &to, nil
+}
 
-	tx := txb.GetTx()
-	fees = tx.GetFee()
+func convertCoins(prices string) (sdk.Coins, error) {
+	coins, err := sdk.ParseCoinsNormalized(prices)
+	if err != nil {
+		return nil, errorsmod.Wrapf(err, "failed to parse coins")
+	}
+	return coins, nil
+}
 
+func (cc *CosmosProvider) signTxBytes(
```

Comment:
this method encode the signed tx, not signing the tx?

Reply:
yes